### PR TITLE
refactor: ♻️ updates logic for disabled icons text

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

 - updates logic for disabled icons text

## Issue

#1430 #1470 

## How to test

1. find a non-DRAFT/PLANNED budget line from Agreement BLI Tab or BudgetLines list
2. hover over row and attempt to delete
3. see new tooltip
4. 💰 profit

## Screenshots

image

https://github.com/user-attachments/assets/cc6feabf-331a-40c8-8d3a-92cab2059e79

## Definition of Done Checklist
- [X] OESA: Code refactored for clarity
- [X] OESA: Dependency rules followed
- [X] Automated unit tests updated and passed
- [X] Automated integration tests updated and passed
- [X] Automated quality tests updated and passed
- [X] Automated load tests updated and passed
- [X] Automated a11y tests updated and passed
- [X] Automated security tests updated and passed
- [X] 90%+ Code coverage achieved
- [-] Form validations updated

### CODE REVIEW

File: frontend/src/hooks/budget-line.hooks.js [hunk 1]
```diff
@@ -9,24 +8,6 @@ import { BLI_STATUS } from "../helpers/budgetLines.helpers";
...
  */
 
-/**
- * This hook returns true if the budget line is in a status that is editable.
- * @param {BudgetLine} budgetLine - The budget line object.
- * @returns {boolean} - Whether the budget line is in a status that is editable.
- * @example
- * const isBudgetLineInEditableStatus = useIsBudgetLineEditableByStatus(budgetLine);
- */
-export const useIsBudgetLineEditableByStatus = (budgetLine) => {
```
Comment (fpigeonjr) [inline frontend/src/hooks/budget-line.hooks.js:19]: not a hook so moved to a helper

File: frontend/src/helpers/budgetLines.helpers.js [hunk 2]
```diff
@@ -127,7 +142,20 @@ export const canLabel = (budgetLine) =>
...
+
+/**
+ * Returns whether the given budget line is editable by status.
+ * @param {BudgetLine} budgetLine - The budget line to check.
+ * @returns {boolean} Whether the budget line is editable by status.
+ **/
+export const isBudgetLineEditableByStatus = (budgetLine) => {
+    const isBudgetLineDraft = budgetLine?.status === BLI_STATUS.DRAFT;
+    const isBudgetLinePlanned = budgetLine?.status === BLI_STATUS.PLANNED;
+    const isBudgetLineInReview = budgetLine?.in_review;
```
Comment (fpigeonjr) [inline frontend/src/helpers/budgetLines.helpers.js:158]: removed EXECUTING budget lines since we only want to allow editing on DRAFT and PLANNED

**Review by maiyerlee — APPROVED** [review]
The tool tip looks great! ✨

**Review by rajohnson90 — APPROVED** [review]

**Comment by fpigeonjr** [comment]
will wait on merge until we run this ux

**Pull request merged**